pkg/schema/load/generic.go

```diff
@@ -155,21 +156,22 @@ func (gvs *genericVersionedSchema) TrimDefaults(r schema.Resource) (schema.Resou
 	return schema.Resource{Value: re}, nil
 }
 
-func removeDefaultHelper(inputdef cue.Value, input cue.Value) (cue.Value, bool, error) {
+func removeDefaultHelper(inputdef cue.Value, input cue.Value) (cue.Value, bool, bool, error) {
```

Comment:
Nit. It's a bit hard to understand what the different return values is/mean `cue.Value, bool, bool, error`. Maybe returning `struct, error` instead could make things easier to reason about/understand?

Reply:
i decided not to remove the empty object, since after it would be difficult to know list[{}, {}] had 2 objects if i removed all empty object.